add-on/panels/js/copy.jsx

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global Components */
+let { utils: Cu } = Components;
+
+var loop = loop || {};
+loop.copy = (mozL10n => {
+  "use strict";
+
+  /**
+   * Copy panel view.
+   */
+  let CopyView = React.createClass({
```

Comment:
You should be able to use the [PureRenderMixin](https://facebook.github.io/react/docs/pure-render-mixin.html) here for a potential slight perf boost.